cfbs/masterfiles/check_download_matches_git.py

```diff
@@ -29,18 +34,64 @@ def check_download_matches_git(versions):
             new_download_dict[key] = value
         download_version_dict = new_download_dict
 
-        with open("differences/difference-" + version + ".txt", "w") as f:
-            only_dl, only_git, value_diff = dict_diff(
-                download_version_dict, git_version_dict
-            )
-
-            print("Files only in the downloaded version:", only_dl, file=f)
-            print("Files only in the git version:", only_git, file=f)
-            print("Files with different contents:", value_diff, file=f)
-
-            if len(only_dl) > 0 or len(value_diff) > 0:
-                user_error(
-                    "Downloadable files of version "
-                    + version
-                    + " do not match git files"
-                )
+        version_diffs_dict = {}
+        version_diffs_dict["files_only_in_downloads"] = []
+        version_diffs_dict["files_only_in_git"] = []
+        version_diffs_dict["files_with_different_content"] = []
+
+        only_dl, only_git, value_diff = dict_diff(
+            download_version_dict, git_version_dict
+        )
+
+        for filepath in only_dl:
+            version_diffs_dict["files_only_in_downloads"].append(filepath)
+        for filepath in only_git:
+            version_diffs_dict["files_only_in_git"].append(filepath)
+        for filepath, _, _ in value_diff:
+            version_diffs_dict["files_with_different_content"].append(filepath)
+
+        diffs_dict["differences"][version] = version_diffs_dict
+
+        if len(only_dl) > 0 or len(value_diff) > 0:
```

Comment:
Here you can consider:
```suggestion
        if only_dl or value_diff:
```
Since an empty list evaluates to `False` and non-empty evaluates to `True`. E.g.:
```Python
>>> bool([])
False
>>> bool(["foo"])
True
```
But you could argue that the former is more readable.

Reply:
Yes, this was on purpose for readability.